controllers/whoToFollow/whoToFollowV2.js

```diff
@@ -0,0 +1,267 @@
+    }
+    //END LOCATIONS
+
+    //OTHER USERS
+    if (page == 1 && topics) {
+      const otherTopicsQuery = `SELECT 
+                            a.user_id,
+                            a.country_code,
+                            a.username,
+                            a.real_name,
+                            a.created_at,
+                            a.updated_at,
+                            a.last_active_at,
+                            a.status,
+                            a.profile_pic_path,
+                            a.profile_pic_asset_id,
+                            a.profile_pic_public_id,
+                            a.bio,
+                            a.is_banned,
+                            a.is_anonymous,
+                            a.encrypted,
+                            a.allow_anon_dm,
+                            a.only_received_dm_from_user_following,
+                            a.is_backdoor_user,
+                            a.followers_count,
+                            a.karma_score,
+                            b.topic_id
+                          FROM users a
+                          INNER JOIN user_topics b ON a.user_id = b.user_id
+                          WHERE 
+                            a.is_anonymous = false
+                            AND a.is_banned = false
+                            AND a.blocked_by_admin = false
+                            AND a.karma_score > 30
+                            AND a.user_id != :admin_user_id
+                            AND b.topic_id NOT IN (:topics)
+                          ORDER BY 
+                            a.last_active_at DESC,
+                            CASE WHEN a.profile_pic_path != '%default-profile-picture%' THEN 0 ELSE 1 END,
+                            a.followers_count DESC
+                          LIMIT :limit`;
+
+      const userOtherTopicFollowerQueryResult = await sequelize.query(otherTopicsQuery, {
+        type: sequelize.QueryTypes.SELECT,
+        replacements: {
+          topics,
+          admin_user_id: process.env.BETTER_ADMIN_ID,
+          limit
+        }
+      });
+      const userOtherTopicFollower = userOtherTopicFollowerQueryResult;
+
+      const tempUsers = [];
+      for (const indexUserOtherTopic in userOtherTopicFollower) {
+        const userTopic = userOtherTopicFollower[indexUserOtherTopic];
+
+        tempUsers.push({
+          ...userTopic,
+          viewtype: 'user'
+        });
+      }
+
+      if (tempUsers.length > 0) {
+        result.push({
+          viewtype: 'labelothers',
+          name: 'Other popular members',
+          users_count: tempUsers.length,
+          users: tempUsers
+        });
+      }
+    }
+    // OTHER USERS
+
+    return res.status(200).json({
+      status: 'success',
+      code: 200,
+      body: result,
+      success: true
+    });
+  } catch (error) {
+    return res.json({
+      success: false,
+      data: 0,
+      message: error?.message
+    });
+  }
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

In the catch block, you are returning a status code of 200 by default, even when an error occurs. This could be misleading as a 200 status code usually indicates that the request was successful. Consider returning an appropriate error status code (like 500 for internal server errors) based on the type of error caught. Also, `data: 0` may not provide meaningful information about the error. Consider providing more descriptive error data.
```diff
- return res.json({
-   success: false,
-   data: 0,
-   message: error?.message
- });
+ return res.status(500).json({
+   success: false,
+   error: error?.message
+ });
```

General:
The SQL queries are written directly in the code, which can lead to potential SQL injection attacks. Consider using parameterized queries or prepared statements to mitigate this risk. Also, it would be better to move these queries to a separate data access layer for better separation of concerns and maintainability.

Lastly, the code lacks error handling for cases when the database queries fail. It's important to handle these errors appropriately to prevent the application from crashing and provide meaningful feedback to the user.


<!-- This is an auto-generated comment by OSS CodeRabbit -->